packages/components/src/internal/components/search/actions.ts

```diff
@@ -15,73 +15,156 @@ import { SchemaQuery } from '../../../public/SchemaQuery';
+    // eslint-disable-next-line prefer-const
+    let { containerPath, requestHandler, ...params } = options;
+
+    let containerPath_: string;
+    if (isAllProductFoldersFilteringEnabled(moduleContext)) {
+        containerPath_ = getProjectPath();
+        params.scope = SearchScope.FolderAndSubfoldersAndShared;
+    } else if (containerPath) {
+        containerPath_ = containerPath;
+    }
+
+    // Return extra info about entity types and material results
+    if (params.experimentalCustomJson === undefined) {
+        params.experimentalCustomJson = true;
+    }
+
+    // Remove the containerID from the returned URL
+    if (params.normalizeUrls === undefined) {
+        params.normalizeUrls = true;
     }
 
-    let containerPath: string;
-    if (isAllProductFoldersFilteringEnabled()) {
-        containerPath = getProjectPath();
-        options.scope = SearchScope.FolderAndSubfoldersAndShared;
+    if (!params.scope) {
+        params.scope = getSearchScopeFromContainerFilter(getContainerFilter(containerPath_));
     }
-    if (filterCategories) {
-        options.category = filterCategories?.join('+');
+
+    if (Array.isArray(params.category)) {
+        (params as any).category = params.category.join('+');
     }
 
     return new Promise((resolve, reject) => {
-        Ajax.request({
+        const request_ = request({
             url: buildURL('search', 'json.api', undefined, {
-                container: containerPath,
+                container: containerPath_,
             }),
-            params: options,
+            method: 'POST',
+            params,
             success: Utils.getCallbackWrapper(json => {
-                addDataObjects(json);
-                const results = new URLResolver().resolveSearchUsingIndex(json);
-                const hits = getProcessedSearchHits(results['hits'], getCardDataFn);
-                resolve({ ...results, hits });
+                if (applyURLResolver) {
+                    resolve(new URLResolver().resolveSearchUsingIndex(json));
+                } else {
+                    resolve(json);
+                }
             }),
-            failure: Utils.getCallbackWrapper(json => reject(json), null, false),
+            failure: handleRequestFailure(reject, 'Failed search query'),
         });
+        requestHandler?.(request_);
     });
+};
+
+export interface SearchHitWithCardData extends SearchHit {
+    cardData: SearchResultCardData;
+}
+
+export interface SearchResultWithCardData extends Omit<SearchResult, 'hits'> {
+    hits: SearchHitWithCardData[];
+}
+
+/** @deprecated Use search() instead */
+export async function searchUsingIndex(
+    options: SearchOptions,
+    getCardDataFn?: GetCardDataFn
+): Promise<SearchResultWithCardData> {
+    const appProps = getPrimaryAppProperties();
+    if (appProps?.productId) {
+        incrementClientSideMetricCount(appProps.productId + 'Search', 'count');
+    }
```

Comment:
I left this metric where it is since it is still utilized by the search page. I could imagine moving the metric to be on the search page specifically. Putting it on the endpoint wrapper will generate tons of counts due to it being used to compose references.